### PR TITLE
Update DeveloperGuide.adoc to mention NameContainsKeywordsPredicate

### PR DESCRIPTION
**Pull request by limmlingg**

### CODE REVIEW

**Comment by Zhiyuan-Amos** [comment]
Actually why do we need to update the Dev guide? :P

**Comment by limmlingg** [comment]
It's difficult to update the Logic class diagram because of the changes that were recently merged.

**Comment by Zhiyuan-Amos** [comment]
No no as in I'm asking, what's the purpose of this PR (i.e why are we including this inside the Developer Guide?)

**Comment by limmlingg** [comment]
Prof Damith requested this;

https://github.com/se-edu/addressbook-level4/pull/431#issuecomment-315581273

**Comment by Zhiyuan-Amos** [comment]
I know that; I'm asking if there's any other reason other than that reason :P Otherwise it's kinda hard to vet this isn't it.

Like just taking a glance, it's pretty random we have a `2.3.1 FindCommand`.

**Comment by yamgent** [comment]
> I know that; I'm asking if there's any other reason other than that reason :P Otherwise it's kinda hard to vet this isn't it.
> 
> Like just taking a glance, it's pretty random we have a 2.3.1 FindCommand.

It is for us to explain any additional details that might not be intuitive at first glance. The design for `FindCommand` now involves an additional new predicate class so it is good to mention about that.

Btw, Prof also mentioned that we should add another section for your Undo & Redo commands. The rationale for doing so is the same (it now involves a stack mechanism).

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -176,6 +176,14 @@ Given below is the Sequence Diagram for interactions within the `Logic` componen
 image::DeletePersonSdForLogic.png[width="800"]
 _Figure 2.3.1 : Interactions Inside the Logic Component for the `delete 1` Command_
 
+==== FindCommand
+
+*API* :
+link:{repoURL}/src/main/java/seedu/address/logic/commands/FindCommand.java[`FindCommand.java`]
+
+. `FindCommand` uses `NameContainsKeywordsPredicate` and passes it to `Model` to filter the list by name.
+. `NameContainsKeywordsPredicate` uses a list of keywords, and returns true if any of the individual words in the person's name matches any of the keywords.
```
Comment (Zhiyuan-Amos) [inline docs/DeveloperGuide.adoc]: `NameContainsKeywordsPredicate` ~~uses~~ **stores** a list of keywords, and returns true if any of the individual words in the person's name matches any of the keywords.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -176,6 +176,14 @@ Given below is the Sequence Diagram for interactions within the `Logic` componen
 image::DeletePersonSdForLogic.png[width="800"]
 _Figure 2.3.1 : Interactions Inside the Logic Component for the `delete 1` Command_
 
+==== FindCommand
+
+*API* :
+link:{repoURL}/src/main/java/seedu/address/logic/commands/FindCommand.java[`FindCommand.java`]
+
+. `FindCommand` uses `NameContainsKeywordsPredicate` and passes it to `Model` to filter the list by name.
```
Comment (Zhiyuan-Amos) [inline docs/DeveloperGuide.adoc]: `to filter the list` -> What list? :P

**Comment by CanIHasReview** [comment]
# v1

@limmlingg submitted v1 for review.

* [v1 1/1] [Update DeveloperGuide\.adoc to mention NameContainsKeywordsPredicate](https://github.com/se-edu/addressbook-level4/pull/562/commits/8f2627d6df8fb83b5d8b58d45683ccfee4133626)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/52ec52609b8a17da202dd3fa8f9813b7820b5d3f...8f2627d6df8fb83b5d8b58d45683ccfee4133626))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/562/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by Zhiyuan-Amos** [comment]
@yamgent what's the status for this? ^^

**Comment by yamgent** [comment]
> @yamgent what's the status for this? ^^

You can continue the work on this PR, as per @damithc's comments, which I can't recall :P.

**Comment by limmlingg** [comment]
Iirc the comment was that we needed to find some documentation or something similar about how predicates are a better choice for filtering lists.

Sorry I haven't been working on this! I've been busy with other work recently.

**Comment by Zhiyuan-Amos** [comment]
> as per @damithc's comments, which I can't recall :P.

Haha I wasn't even there when prof gave the comments ><

> Sorry I haven't been working on this! I've been busy with other work recently.

I think it's ok if you don't finish this. You can raise an issue on this and link it to your PR so someone else working on this issue can refer to this PR. Also, put this in the issue description as it'll be helpful ^^ "we needed to find some documentation or something similar about how predicates are a better choice for filtering lists."

**Comment by Zhiyuan-Amos** [comment]
```
updateFilteredPersonListByName(Set<String>)
updateFilteredPersonListByPhone(Set<String>)
updateFilteredPersonListByEmail(Set<String>)
...

vs

updateFilteredPersonList(Predicate<ReadOnlyPerson>)

When future Predicates are implemented e.g. 
PhoneContainsKeywordsPredicate, these Predicates will also implement
Predicate<ReadOnlyPerson>. As such, the Model can filter by having
1 method only.
```

Please rephrase it; typed it in a rush :P

**Comment by Zhiyuan-Amos** [comment]
@limmlingg gentle reminder ^^

**Pull request closed without merging**